Remotion/Data/DomainObjects/Queries/QueryDefinitionRepository.cs

```diff
@@ -34,21 +34,33 @@ public static QueryDefinitionRepository FromQueryDefinitions (IEnumerable<QueryD
     {
       ArgumentUtility.CheckNotNull(nameof(queryDefinitions), queryDefinitions);
 
-      return new QueryDefinitionRepository(queryDefinitions.ToDictionary(e => e.ID, e => e));
+      return new QueryDefinitionRepository(queryDefinitions);
     }
 
     private readonly Dictionary<string, QueryDefinition> _queryLookup;
 
     public QueryDefinitionRepository (IQueryDefinitionLoader queryDefinitionLoader)
+        : this(ArgumentUtility.CheckNotNull(nameof(queryDefinitionLoader), queryDefinitionLoader).LoadAllQueryDefinitions())
     {
-      ArgumentUtility.CheckNotNull(nameof(queryDefinitionLoader), queryDefinitionLoader);
-
-      _queryLookup = queryDefinitionLoader.LoadAllQueryDefinitions().ToDictionary(e => e.ID, e => e);
     }
 
-    private QueryDefinitionRepository (Dictionary<string, QueryDefinition> queryLookup)
+    private QueryDefinitionRepository (IEnumerable<QueryDefinition> queryDefinitions)
     {
-      _queryLookup = queryLookup;
+      ArgumentUtility.CheckNotNull(nameof(queryDefinitions), queryDefinitions);
+
+      try
+      {
+        _queryLookup = queryDefinitions.ToDictionary(e => e.ID, e => e);
+      }
+      catch (NullReferenceException)
+      {
+        // Only way a null ref is causes is in the key selector of the .ToDictionary(...) call, indicating that an item is null
```

Comment:
These messages do not tell the caller which query is the problem, which makes fixing it harder. We should provide the ID of the QueryDefinition in the message, so that text search for the source of the problem is possible.
This may necessitate manually building the dictionary.

We could even strive to name not only the first ID with an error, but all IDs, so that fixing the problem does not require multiple attempts.